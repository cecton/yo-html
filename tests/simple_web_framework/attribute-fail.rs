pub mod basic_web_framework;

use basic_web_framework::prelude::*;

fn invalid_token_in_attr_value() {
    let _ = html! { <span class=* /> };
}

fn invalid_token_in_attr_name() {
    let _ = html! { <span ()=42 /> };
}

fn missing_value() {
    let _ = html! { <span class= /> };
    let _ = html! { <span {class}= /> };
}

fn missing_eq() {
    let _ = html! { <span class /> };
}

fn duplicate_attributes() {
    let class = "46";
    let _ = html! { <span class="42" class="43" class=["44", "45"] {class} /> };
}

fn invalid_shorthand() {
    let _ = html! { <span {"class"} /> };
}

fn main() {}
