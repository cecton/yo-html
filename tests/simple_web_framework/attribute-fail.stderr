error: missing value in attribute
 --> tests/simple_web_framework/attribute-fail.rs:6:27
  |
6 |     let _ = html! { <span class=* /> };
  |                           ^^^^^

error: expected identifier, block or shorthand property
  --> tests/simple_web_framework/attribute-fail.rs:10:27
   |
10 |     let _ = html! { <span ()=42 /> };
   |                           ^

error: missing value in attribute
  --> tests/simple_web_framework/attribute-fail.rs:14:27
   |
14 |     let _ = html! { <span class= /> };
   |                           ^^^^^

error: missing value in attribute
  --> tests/simple_web_framework/attribute-fail.rs:15:27
   |
15 |     let _ = html! { <span {class}= /> };
   |                           ^^^^^^^

error: missing `=` token in attribute
  --> tests/simple_web_framework/attribute-fail.rs:19:27
   |
19 |     let _ = html! { <span class /> };
   |                           ^^^^^

error: attribute already defined
  --> tests/simple_web_framework/attribute-fail.rs:24:38
   |
24 |     let _ = html! { <span class="42" class="43" class=["44", "45"] {class} /> };
   |                                      ^^^^^

error: attribute already defined
  --> tests/simple_web_framework/attribute-fail.rs:24:49
   |
24 |     let _ = html! { <span class="42" class="43" class=["44", "45"] {class} /> };
   |                                                 ^^^^^

error: attribute already defined
  --> tests/simple_web_framework/attribute-fail.rs:24:69
   |
24 |     let _ = html! { <span class="42" class="43" class=["44", "45"] {class} /> };
   |                                                                     ^^^^^

error: expected identifier
  --> tests/simple_web_framework/attribute-fail.rs:28:28
   |
28 |     let _ = html! { <span {"class"} /> };
   |                            ^^^^^^^
